.github/workflows/build.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2022, Kry10 Limited
+# SPDX-License-Identifier: BSD-2-Clause
+
+name: Build
+
+on:
+  pull_request:
+    # Currently aspirational
+    branches:
+      - master
+  push:
+    branches:
+      - ci-riscv64
+  schedule:
+    - cron: '0 17 * * 6'
```

Comment:
Better to choose a non-0 random minute for cron jobs on larger infrastructure (because lots of jobs will start there, the random minute helps to avoid those load points and get smoother startup etc)

Reply:
Actually, it's not clear to me whether this schedule will run. I didn't see it run over the weekend.

The [docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) say "scheduled workflows run on the latest commit on the default or base branch". I had initially thought that just meant that `github.ref` pointed to the default branch, but I'm now thinking that it means the workflow also needs to be defined on the default branch.

The schedule is not that important right now, so I might just take this out. It only really becomes important when graph-refine is stable enough that we can leave it alone for extended periods. I imagine graph-refine will need some level of hand-holding for a while.

Reply:
Oh, you're completely right, the workflow does have to be on the default branch. In case of emergency, you could add the workflow file to the default branch, but explicitly check out the ci branch as the branch to be worked on for scheduled events. It does duplicate the workflow file over branches, but at least in a way that should be relatively easy for cherrypicking commits on it etc.